pkg/option/config.go

```diff
@@ -15,71 +15,123 @@
-type MapOptions struct {
-	vals      map[string]string
-	validator Validator
-}
+	// AllowLocalhostAlways always allows the local stack to reach local
+	// endpoints
+	AllowLocalhostAlways = "always"
+
+	// AllowLocalhostPolicy requires a policy rule to allow the local stack
+	// to reach particular endpoints or policy enforcement must be
+	// disabled.
+	AllowLocalhostPolicy = "policy"
+
+	// ModePreFilterNative for loading progs with xdpdrv
+	ModePreFilterNative = "native"
+
+	// ModePreFilterGeneric for loading progs with xdpgeneric
+	ModePreFilterGeneric = "generic"
+)
+
+// Config is the configuration used by Daemon.
+type Config struct {
+	BpfDir          string     // BPF template files directory
+	LibDir          string     // Cilium library files directory
+	RunDir          string     // Cilium runtime directory
+	NAT46Prefix     *net.IPNet // NAT46 IPv6 Prefix
+	Device          string     // Receive device
+	DevicePreFilter string     // XDP device
+	ModePreFilter   string     // XDP mode, values: { native | generic }
+	HostV4Addr      net.IP     // Host v4 address of the snooping device
+	HostV6Addr      net.IP     // Host v6 address of the snooping device
+	IPv4Disabled    bool       // Disable IPv4 allocation
+	LBInterface     string     // Set with name of the interface to loadbalance packets from
+
+	Tunnel string // Tunnel mode
+
+	DryMode       bool // Do not create BPF maps, devices, ..
+	RestoreState  bool // RestoreState restores the state from previous running daemons.
+	KeepConfig    bool // Keep configuration of existing endpoints when starting up.
+	KeepTemplates bool // Do not overwrite the template files
+
+	// AllowLocalhost defines when to allows the local stack to local endpoints
+	// values: { auto | always | policy }
+	AllowLocalhost string
+
+	// StateDir is the directory where runtime state of endpoints is stored
+	StateDir string
+
+	// Options changeable at runtime
+	Opts *BoolOptions
 
-// NamedMapOptions is a MapOptions struct with a configuration name.
-// This struct is useful to keep reference to the assigned
-// field name in the internal configuration struct.
-type NamedMapOptions struct {
-	name string
-	MapOptions
+	// Mutex for serializing configuration updates to the daemon.
+	ConfigPatchMutex lock.RWMutex
+
+	// Monitor contains the configuration for the node monitor.
+	Monitor *models.MonitorStatus
+
+	// AccessLog is the path to the access log of supported L7 requests observed.
+	AccessLog string
+
+	// AgentLabels contains additional labels to identify this agent in monitor events.
+	AgentLabels []string
 }
 
-// NewNamedMapOptions creates a reference to a new NamedMapOpts struct.
-func NewNamedMapOptions(name string, values *map[string]string, validator Validator) *NamedMapOptions {
-	return &NamedMapOptions{
-		name:       name,
-		MapOptions: *NewMapOpts(*values, validator),
-	}
+var (
+	config *Config
+	once   sync.Once
+)
+
+// GetConfig returns the global configuration set in flags
```

Comment:
This should state that this is the global daemon config singleton. I only knew that because I knew the context of the PR. Unfortunately, naming it `GetDaemonConfig` would make it annoyingly long and make lines wider, but that would also be good.